### DISCUSSION TITLE
Is package.json required for plugins to work?

### DISCUSSION THREAD

**Discussion by HenrikPoulsen** [semantic-release/semantic-release#3082] [Q&A] [answered]
I need to use the gitlab plugin.
So I tried installing it globally with `npm i -g @semantic-release/gitlab` before running `npx semantic-release` and it cannot find the plugin.
```
[semantic-release] » ✘  An error occurred while running semantic-release: Error: Cannot find module '@semantic-release/gitlab'
```
If I do install it with `-D` it generates a package.json file in the repo and then it can load the plugin just fine.

Since this is not an npm based repository I would prefer to not have a package.json file in the repo. Is there any way to get it to load the plugins without a package.json present?

**Comment by sheerlox** [comment, marked as answer]
`npm install --no-save --prefix . @semantic-release/gitlab` should do the trick :wink:

**Reply by HenrikPoulsen** [reply]
Learned something new today. Thanks 🙂

**Reply by sheerlox** [reply]
Learned that yesterday, had to solve the same issue for Elixir projects :sweat_smile:
